src/components/AppContainer.js

```diff
@@ -1,11 +1,194 @@
-import React, { Component } from 'react'
+import React, { Component } from 'react';
+import axios from 'axios';
+import ComponentPostarProduto from './PostarProduto/ComponentPostarProduto';
+import ComponentFiltro from './ComponentFiltro/Index';
+import Footer from "./Footer/Index";
+import Header from "./CardHeader/Index";
+import CardCategoria from './CardCategoria/CardCategoria';
+import PaginaCategoria from './CardCategoria/PaginaCategoria';
+import Carrinho from './Carrinho/Carrinho'
+import DeletaProduto from './DeletaProduto/DeletaProduto'
+import { baseUrl } from '../constants/index.js';
+import {NumeroCarrinho} from './styles'
 
 export class AppContainer extends Component {
+  state = {
+    paginaAtual: "Inicio",
+    valorInputMin: "",
+    valorInputMax: "",
+    valorCategoria: "",
+    valorInputOrdenacao: "",
+    listaDeProdutos: [],
+    produtosSelecionados: [],
+    total: 0
+  }
+  componentDidMount = () => {
+    this.listarProdutos();
+  }
+  onChangeOrdenacao = (e) => {
+    this.setState({valorInputOrdenacao: e.target.value})
+    switch (e.target.value) {
+      case 'nome':
+        this.setState({listaDeProdutos: this.state.listaDeProdutos.sort((a, b) => {
+          if(b.name > a.name) {
+            return -1;
+          }
+          return true
+        })})
+        break;
+      case 'valor':
+        this.setState({listaDeProdutos: this.state.listaDeProdutos.sort((a, b) => {
+          return a.price - b.price
+          })})
+        break;  
+      default:
+        return this.listarProdutos()
+    }
+  }
+  onChangeValorMin = (e) => {
+    this.setState({valorInputMin: e.target.value})
+  }
+  onChangeValorMax = (e) => {
+    this.setState({valorInputMax: e.target.value})
+  }
+  onChangeValorCategoria = (e) => {
+    this.setState({valorCategoria: e.target.value})
+  }
+  listarProdutos = () => {
+    axios.get(baseUrl)
+    .then( response => {
+      this.setState({ listaDeProdutos: response.data.products });
+    })
+    .catch( err => {
+      alert(err.message);
+    })
+  }
+  mudaPagina = (a) => {
+    switch(a) {
+      case "Vender":
+        this.setState({ paginaAtual: "Vender"});
+        break;
+      case "Carrinho":
+        this.setState({ paginaAtual: "Carrinho"});
+        break;  
+      case "Roupas":
+        this.setState({ paginaAtual: "Roupas"});
+        break;  
+      case "Calcados":
+        this.setState({ paginaAtual: "Calcados"});
+        break;  
+      case "Acessórios":
+        this.setState({ paginaAtual: "Acessórios"});
+        break;  
+      case "Outros":
+        this.setState({ paginaAtual: "Outros"});
+        break;  
+      default:
+        this.setState({ paginaAtual: "Inicio" });
+    }
```

Comment:
Acho que dava para simplificar isso aqui hein. Se vcs estão verifcando o que é o parâmetro e setando no estado o exato valor do que tão verificando, bastaria setar o parâmetro diretamente: `this.setState({paginaAtual: a})`

Além disso, nomeem melhor os seus parâmetros de funções. Fiquei algum tempo para descobrir o que é um parâmetro `a`?